test/llvm-intrinsics/bitreverse_small_type.ll

```diff
@@ -0,0 +1,38 @@
+;; Check that llvm.bitreverse.* intrinsics are lowered to emulation functions for
+;; 2/4-bit types.
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv -spirv-ext=+SPV_INTEL_arbitrary_precision_integers -spirv-text %t.bc -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv -spirv-ext=+SPV_INTEL_arbitrary_precision_integers %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o - | llvm-dis -o - | FileCheck %s --check-prefix=CHECK-LLVM
+
+; There is no validation for SPV_INTEL_arbitrary_precision_integers implemented in
+; SPIRV-Tools. TODO: spirv-val %t.spv
+
+; SPV_KHR_bit_instructions extension was not enabled so BitReverse must not be generated
+; CHECK-SPIRV-NOT: BitReverse
```

Comment:
is it possible to add more SPIR-V checks here?

Reply:
Added checks for lowered function names.